hkm/finna.py

```diff
@@ -299,5 +314,20 @@ def get_labeled_high_res_url(self, record, label):
 
         return None
 
+    def get_identifier(self, record):
+        if not (images_extended := record.get("imagesExtended")):
+            return None
+
+        LOG.debug(
+            "Found images_extended",
+            extra={"data": {"result_data": images_extended}},
+        )
+
+        for image_extended in images_extended:
+            if identifier := image_extended.get("identifier"):
+                return identifier
```

Comment:
This return from inside a loop looks wierd, but I think it's totally acceptable in Python world 👌